pkg/media/manifest_builder.go

```diff
@@ -0,0 +1,177 @@
+package media
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"git.stream.place/streamplace/c2pa-go/pkg/c2pa"
+	"stream.place/streamplace/pkg/aqtime"
+	"stream.place/streamplace/pkg/log"
+	"stream.place/streamplace/pkg/model"
+	"stream.place/streamplace/pkg/streamplace"
+)
+
+// ManifestBuilder is responsible for creating C2PA (Content Credentials) manifests
+// for livestream segments.
+// The builder creates manifests that include:
+// - Basic livestream information (title, creator, date)
+// - Content rights and copyright information
+// - Content warnings for sensitive material
+// - Distribution policies
+// - C2PA action history (created, published)
+// The manifest is meant to align closely with the IPTC Video Metadata Recommendations.
+// See https://iptc.org/std/videometadatahub/recommendation/IPTC-VideoMetadataHub-props-Rec_1.6.html
+type ManifestBuilder struct {
+	model model.Model
+}
+
+func NewManifestBuilder(model model.Model) *ManifestBuilder {
+	return &ManifestBuilder{
+		model: model,
+	}
+}
+
+func (mb *ManifestBuilder) BuildManifest(ctx context.Context, streamerName string, start int64) (*c2pa.ManifestDefinition, error) {
+	// Start with base manifest
+	mani := obj{
+		"title": fmt.Sprintf("Livestream Segment at %s", aqtime.FromMillis(start)),
+		"assertions": []obj{
+			{
+				"label": "c2pa.actions",
+				"data": obj{
+					"actions": []obj{
+						{"action": "c2pa.created"},
+						{"action": "c2pa.published"},
+					},
+				},
+			},
+			{
+				"label": StreamplaceMetadata,
+				"data": obj{
+					"@context": obj{
+						"dc": "http://purl.org/dc/elements/1.1/",
+						"Iptc4xmpExt": "http://iptc.org/std/Iptc4xmpExt/2008-02-29/",
+						"photoshop": "http://ns.adobe.com/photoshop/1.0/",
+						"xmpRights": "http://ns.adobe.com/xap/1.0/rights/",
+					},
+					"dc:creator": streamerName,
+					// TODO: Add the title of the livestream. This should come from the livestream record.
+					"dc:title":   []string{"livestream"},
+					"dc:date":    []string{aqtime.FromMillis(start).String()},
+				},
+			},
+		},
+	}
+
+	// Add database metadata if available
+	if mb.model != nil {
+		metadata, err := mb.model.GetDefaultMetadata(ctx, streamerName)
+		if err != nil {
+			log.Warn(ctx, "ManifestBuilder: failed to retrieve metadata, using defaults", "error", err, "did", streamerName)
```

Comment:
Our convention here is that `model` methods that retrieve one record return `nil, nil` when that record doesn't exist. So if we got an error it's something worse than "metadata not found" and it'd be appropriate for us to error out:

```suggestion
			return nil, fmt.Errorf("failed to retrieve metadata: %w", err)
```